ctapipe/instrument/tests/test_trigger.py

```diff
@@ -6,6 +8,17 @@
 from ctapipe.io import EventSource
 
 
+def assert_all_tel_keys(event, expected):
+    expected = tuple(expected)
+    for name, container in event.items():
+        if hasattr(container, "tel"):
+            actual = tuple(container.tel.keys())
+            if len(actual) > 0 and actual != expected:
```

Comment:
Not sure what the `len(actual)`  test is for? Empty tuples seem to evaluate as identical, so what are you testing for here?

Reply:
data levels that are not filled at all (e.g. r0, dl2) would compare false to a list of expected telescopes

Reply:
If you want to skip this test for certain data levels then demanding a certain length seems like a very non-obvious way to test for that. Can't you put the test for if a DL is filled into the `hasattr` conditional?

Reply:
I added a keyword to explicitly ignore certain sub-containers instead of checking length 0